core/indexer/common.go

```diff
@@ -367,62 +367,26 @@ func serializeAccountsHistory(accounts map[string]*AccountBalanceHistory) []byte
 func prepareSerializedDataForATransaction(
 	tx *Transaction,
 	selfShardID uint32,
-	isMBOfTxInDB bool,
+	_ bool,
 ) (meta []byte, serializedData []byte) {
-	var err error
-	if isMBOfTxInDB {
-		if !isCrossShardDstMe(tx, selfShardID) || tx.Status == transaction.TxStatusInvalid.String() {
-			return
-		}
-		// update transaction
-		meta, serializedData = prepareTxUpdate(tx)
-	} else {
-		// insert transaction in database
-		meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, tx.Hash, "\n"))
-		serializedData, err = json.Marshal(tx)
-		if err != nil {
-			log.Debug("indexer: marshal",
-				"error", "could not serialize transaction, will skip indexing",
-				"tx hash", tx.Hash)
-			return
-		}
-	}
-	return
-}
+	meta = []byte(fmt.Sprintf(`{"update":{"_id":"%s", "_type": "_doc"}}%s`, tx.Hash, "\n"))
 
-func prepareSerializedAccountInfo(address string, account *AccountInfo) (meta []byte, serializedData []byte) {
-	var err error
-	meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, address, "\n"))
-	serializedData, err = json.Marshal(account)
+	marshalledTx, err := json.Marshal(tx)
 	if err != nil {
 		log.Debug("indexer: marshal",
-			"error", "could not serialize account, will skip indexing",
-			"address", address)
+			"error", "could not serialize transaction, will skip indexing",
+			"tx hash", tx.Hash)
 		return
 	}
-
-	return
-}
-
-func prepareSerializedAccountBalanceHistory(address string, account *AccountBalanceHistory) (meta []byte, serializedData []byte) {
-	var err error
-	meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, address, "\n"))
-	serializedData, err = json.Marshal(account)
-	if err != nil {
-		log.Debug("indexer: marshal",
-			"error", "could not serialize account history entry, will skip indexing",
-			"address", address)
+	if !isCrossShardDstMe(tx, selfShardID) || tx.Status == transaction.TxStatusInvalid.String() {
+		serializedData =
+			[]byte(fmt.Sprintf(`{"script":{"source":""},"upsert":%s}`,
+				string(marshalledTx)))
+		log.Warn("log meta - insert", "data", string(meta))
+		log.Warn("log srld data - insert", "data", string(serializedData))
```

Comment:
what is srld? why do we have 2 log.Warn calls?

Reply:
removed